WalletWasabi.Fluent/Models/WalletBackup.cs

```diff
@@ -0,0 +1,22 @@
+using NBitcoin;
+using WalletWasabi.Blockchain.Keys;
+using WalletWasabi.Wallets.Slip39;
+
+namespace WalletWasabi.Fluent.Models;
+
+public abstract record WalletBackup(string? Password = "");
+
+public record RecoveryWordsBackup(
+	string? Password = "",
+	Mnemonic? Mnemonic = null) : WalletBackup(Password);
+
+public record MultiShareBackupSettings(
+	byte Threshold = KeyManager.DefaultShamirThreshold,
+	byte Shares = KeyManager.DefaultShamirShares);
+
+public record MultiShareBackup(
+	MultiShareBackupSettings Settings,
+	string? Password = "",
+	Share[]? Shares = null,
+	byte CurrentShare = 0,
+	byte CurrentSharePage = 0) : WalletBackup(Password);
```

Comment:
I am not sure any of these fields could be really null in any case. Why would we create a `MultiShareBackup` with a null `Shares`? I think never. The same is valid for `Password` and `Mnemonics`.  The problem here is not that it forces to validate states that can never be invalid everywhere but that the types suggest that you can have those invalid states somewhere when you really cannot.

Reply:
https://github.com/WalletWasabi/WalletWasabi/pull/13800/commits/c1c86c43f1f8d0bc7c46ef0efe507e9cd60f78e7